bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java

```diff
@@ -265,10 +268,18 @@ public BigtableSession(BigtableOptions opts) throws IOException {
 
     BigtableClientMetrics.counter(MetricLevel.Info, "sessions.active").inc();
 
-    // Defer the creation of both the tableAdminClient until we need them.
-    //TODO(rahulkql): Need to identify which resources to initialized for GCJ's adapter case.
-    this.adminSettings = BigtableVeneerSettingsFactory.createTableAdminSettings(options);
-    this.baseAdminSettings = BaseBigtableTableAdminSettings.create(adminSettings.getStubSettings());
+
+    //TODO(rahulkql): Need to identify which resources to initialized for GCJ's adapter case
+    if(options.useGCJClient()){
```

Comment:
@sduskis 
Have included `dataSettings` & `dataClient` creation in xor of `BigtableSession` only in case of `options.useGCJClient.` 
As moving it to getClientWrapper would introduce a chain of updates to handle `IOException`.

If needed, I could create a separate PR on which we can move these into `BigtableSession#getClientWrapper` & `BigtableSession#getAdminClientWrapper`.

Reply:
Please wrap the `IOException` in a `RuntimeException` for now.